driver-core/src/main/java/com/datastax/driver/core/TableOptionsMetadata.java

```diff
@@ -188,7 +188,12 @@
     if (is380OrHigher) this.cdc = isNullOrAbsent(row, CDC) ? DEFAULT_CDC : row.getBool(CDC);
     else this.cdc = DEFAULT_CDC;
 
-    this.scyllaCDC = this.extensions.containsKey(SCYLLA_CDC_EXTENSION);
+    ByteBuffer rawScyllaCDC = this.extensions.get(SCYLLA_CDC_EXTENSION);
```

Comment:
Could this not be generalized to something like 
```
for (key : MAP_EXTENSION_NAMES) {
    var ext = this.extensions.get(key);
   if (ext != null) {
      myMapExtensions.put(key, ...parse...);
}
}
```
And similarly, in the print, for (mapext: mapexts) { print(key, map)... yada yada.
Then this would also handle encryption at rest and whatever else we throw at it, just by adding the key name to a set.

Reply:
In the amended commit, I generalized this implementation and also added support for other Scylla "map" extensions (Alternator Tags, Encryption at Rest). Tested all those extensions manually: they parse correctly and generated CQL by `toString()` is correct for all those cases.